src/counts.py

```diff
@@ -8,6 +8,12 @@
 # Count the characters in `x`` and put the counts in `counts`.
 # Your code goes here.
 
+for c in x: 
+    if c in count:
+        count[c] += 1
+    if c not in count:
+        count[c]=1
```

Comment:
Excellent solution.

This would also have worked:

```python
for c in x:
    if c not in counts:
        counts[c] = 0 # add it with count zero; we will increment to one shortly
    counts[c] += 1
```

There is also a builtin counter in Python (but that was not the intended solution, and you did the right thing).

```python
>>> from collections import Counter
>>> Counter('foobar')
Counter({'o': 2, 'f': 1, 'b': 1, 'a': 1, 'r': 1})
```

It isn't always the most convenient, though, so it is good to know how to roll your own, as you just did.